dastardcommander/trigger_blocker.py

```diff
@@ -0,0 +1,109 @@
+import json
+import os
+
+
+class TriggerBlocker:
+    """
+    Handle a more-or-less permanent list of channels that are considered "bad" and should be
+    blocked from triggering no matter what the user asks for. That is, any "all channels" settings
+    will not allow triggering of these channels.
+
+    Values are stored to and read from a config file (default: ~/.dastard/blocked_channels.json),
+    including a history of the last several lists.
+
+    Usage:
+    >>> tb = TriggerBlocker(tempfile.mkstemp(suffix=".json")[1])
+    >>> tb.block_channels(6,4,2)
+    >>> print(tb.blocked)
+    [2, 4, 6]
+    >>> tb.block_channels(8,6,[4,2],4,6,8)
+    >>> print(tb.blocked)
+    [2, 4, 6, 8]
+    >>> tb.unblock_channels(4)
+    >>> print(tb.blocked)
+    [2, 6, 8]
+    >>> tb.revert_prev()
+    >>> print(tb.blocked)
+    [2, 4, 6, 8]
+    >>> tb.clear()
+    >>> print(tb.blocked)
+    []
+    """
+
+    FIELDS_TO_MEMO = ("blocked", "blocked_history")
+    HISTORY_LENGTH = 10
+
+    def __init__(self, configfile=None):
+        if configfile is None:
+            configfile = "~/.dastard/blocked_channels.json"
```

Comment:
I'm pretty sure you can just put the json string into a `QSettings` so we don't mix and match ways to save data..... though I guess having the json on disk does open up some automation options.

Reply:
`QSettings` would probably have been easier. But it's been pretty nice to debug the JSON file problems, because I can just look at the darn file (and change it, when needed).